docs/Add-Models.md

```diff
@@ -0,0 +1,270 @@
+
+### Write Config
+We start by writing your model config class. This class will register all the hyperparameters and axes of your models. We want to define them as the first step because they will be used immidately in the next step that implements your model.
+
+Note that you don't need to write all of configurations at once. You can start with the key hyperparameters and axes, and add more as you implement the model.
+
+**Hyperparameters** should be declared with their type and a default value, illustrated below:
+
+```python
+seq_len: int = 2048
+hidden_dim: int = 4096
+intermediate_dim: int = 11008
+```
+
+**Model axies** are used for parallelization. An Axis is registered with its name and size. The size of an Axis is normally associated with a hyperparameter. For example:
+
+```python
+Pos = property(lambda self: Axis(name="position", size=self.seq_len))
+Embed = property(lambda self: Axis(name="embed", size=self.hidden_dim))
+Mlp = property(lambda self: Axis(name="mlp", size=self.intermediate_dim))
+```
+
+For real examples and deeper understanding, check out `Gpt2Config` in [gpt2.py](https://github.com/stanford-crfm/levanter/blob/main/src/levanter/models/gpt2.py) and `LlamaConfig` in [llama.py](https://github.com/stanford-crfm/levanter/blob/main/src/levanter/models/llama.py).
+
+To convert your config class to and from Hugging Face config class, you can write class functions `to_hf_config()` and `from_hf_config()` that maps the input parameters to the corresponding Hugging Face parameters. For example, in Llama, we have the following:
+
+```python
+from transformers import PretrainedConfig as HfPretrainedConfig
+
+# ...
+
+@classmethod
+def from_hf_config(cls, hf_config: HfPretrainedConfig):
+    return LlamaConfig(
+        seq_len=hf_config.max_position_embeddings,
+        hidden_dim=hf_config.hidden_size,
+        intermediate_dim=hf_config.intermediate_size,
+        num_layers=hf_config.num_hidden_layers,
+        num_heads=hf_config.num_attention_heads,
+        activation_function=hf_config.hidden_act,
+        initializer_range=hf_config.initializer_range,
+        layer_norm_epsilon=hf_config.rms_norm_eps,
+        rope_scaling=hf_config.rope_scaling,
+    )
+```
+
+Lastly, you should register your head model's class name as a class property. This step can be deferred until the head class is constructed.
+This class property would make it easier to call your model with the config class. For example:
+
+```python
+@property
+def model_type(cls) -> Type["LlamaLMHeadModel"]:
+    return LlamaLMHeadModel
+```
+
+### Write Model
+After you have defined your config class, you can start writing your model.
+You can follow the breakdown of the model in your Hugging Face implmentation. This would make it easier to validate the implementation through unit tests.
```

Comment:
```suggestion
You can follow the breakdown of the model in your Hugging Face implementation. This would make it easier to validate the implementation through unit tests.
```